File: C++/problem0922.cpp
#include <iostream>
#include <iomanip>
#include <vector>
#include <map>
#include <set>
#include <algorithm>
#include <cmath>
#include <ctime>
#include <cstdlib>
#include <windows.h>
#include <string>
#include <numeric>

using namespace std;

class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& A) {
        int j = 1;
        for (int i = 0; i < A.size(); i += 2)
        {
            if (A[i]%2 == 1)
            {
                while (A[j]%2 == 1)
                    {
                        j += 2;
                    }
                int tmp = A[i];
                A[i] = A[j];
                A[j] = tmp;
            }
        }
        return A;
    }
};

int main()
{
    vector<int> A {4,2,5,7};
    Solution solu;
    vector<int> ans = solu.sortArrayByParityII(A);
    vector<int>::iterator it;
    for (it = ans.begin(); it < ans.end(); ++it)
    {
        cout << *it << '\t';
    }
    cout << endl;
    return 0;
}

File: Python/problem0922.py
from typing import List

class Solution:
    def sortArrayByParityII(self, A: List[int]) -> List[int]:
#         OddInEvenInds, EvenInOddInds = [], []
#         for k in range(0, len(A), 2):
#             if A[k]%2 == 1:
#                 OddInEvenInds.append(k)
#         for k in range(1, len(A), 2):
#             if A[k]%2 == 0:
#                 EvenInOddInds.append(k)
#         for i in range(len(OddInEvenInds)):
#             A[OddInEvenInds[i]], A[EvenInOddInds[i]] = A[EvenInOddInds[i]], A[OddInEvenInds[i]]
#         return A

        # odd, even = [], []
        # for a in A:
        #     if a % 2 == 0:
        #         even.append(a)
        #     else:
        #         odd.append(a)
        # for i in range(len(A)):
        #     if i % 2 == 0:
        #         A[i] = even.pop()
        #     else:
        #         A[i] = odd.pop()
        # return A

        # ans = [0] * len(A)
        # i, j = 0, 1
        # for a in A:
        #     if a % 2 == 0:
        #         ans[i] = a
        #         i += 2
        #     else:
        #         ans[j] = a
        #         j += 2
        # return ans

        j = 1
        for i in range(0, len(A), 2):
            if A[i] % 2 == 1:
                while A[j] % 2 == 1:
                    j += 2
                A[i], A[j] = A[j], A[i]
        return A      




